x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -0,0 +1,495 @@
+      return sort ? sort.map((sortingCol) => [sortingCol.columnId, sortingCol.sortDirection]) : [];
+    }, [sort]);
+
+    const onSort = useCallback(
+      (nextSort: string[][]) => {
+        dispatch(
+          timelineActions.updateSort({
+            id: timelineId,
+            sort: nextSort.map(([id, direction]) => {
+              const currentColumn = columns.find((column) => column.id === id);
+              const columnType = currentColumn ? currentColumn.type : 'keyword';
+              return {
+                columnId: id,
+                columnType,
+                sortDirection: direction,
+              } as SortColumnTimeline;
+            }),
+          })
+        );
+      },
+      [dispatch, timelineId, columns]
+    );
+
+    const setAppState = useCallback(
+      (newState: { columns: string[]; sort?: string[][] }) => {
+        if (newState.sort) {
+          onSort(newState.sort);
+        } else {
+          const columnsStates = newState.columns.map((columnId) =>
+            getUnifiedDataTableColumnHeader(columnId, defaultUdtHeaders)
+          );
+          dispatch(timelineActions.updateColumns({ id: timelineId, columns: columnsStates }));
+        }
+      },
+      [dispatch, onSort, timelineId]
+    );
+
+    // Columns management
+    const { onSetColumns } = useUnifiedTableColumns({
+      capabilities,
+      dataView,
+      dataViews,
+      setAppState,
+      useNewFieldsApi: true,
+      columns: defaultColumnIds,
+      sort: sortingColumns,
+    });
+
+    const onColumnResize = useCallback(
+      ({ columnId, width }: { columnId: string; width: number }) => {
+        dispatch(
+          timelineActions.updateColumnWidth({
+            columnId,
+            id: timelineId,
+            width, // initialWidth?
+          })
+        );
+      },
+      [dispatch, timelineId]
+    );
+
+    const onResizeDataGrid = useCallback(
+      (colSettings) => {
+        onColumnResize({ columnId: colSettings.columnId, width: Math.round(colSettings.width) });
+      },
+      [onColumnResize]
+    );
+
+    const onAddFilter = useCallback(
+      (field: DataViewField | string, values: unknown, operation: '+' | '-') => {
+        if (dataView && filterManager) {
+          const fieldName = typeof field === 'string' ? field : field.name;
+          popularizeField(dataView, fieldName, dataViews, capabilities);
+          const newFilters = generateFilters(filterManager, field, values, operation, dataView);
+          return filterManager.addFilters(newFilters);
+        }
+      },
+      [filterManager, dataView, dataViews, capabilities]
+    );
+
+    const onChangeItemsPerPage = useCallback(
+      (itemsChangedPerPage) =>
+        dispatch(
+          timelineActions.updateItemsPerPage({ id: timelineId, itemsPerPage: itemsChangedPerPage })
+        ),
+      [dispatch, timelineId]
+    );
+
+    const browserFieldsByName = useMemo(() => getAllFieldsByName(browserFields), [browserFields]);
```

Comment:
I wonder how often we do this in timeline. Is it something that you think should live in redux...An eventual `sourcerer` reducer?

Reply:
I think we should always only have the source data in redux, then use selectors to process/modify and retrieve on the fly

Reply:
This is not needed and has now been removed.